psi4/driver/json_wrapper.py

```diff
@@ -32,27 +32,202 @@
+    elif json_data["driver"] == "properties":
+        ret = {}
+        mtd = json_data["model"]["method"].upper()
+        if "dipole" in kwargs["properties"]:
+            ret["dipole"] = [psi_props[mtd + " DIPOLE " + x] for x in ["X", "Y", "Z"]]
+        if "quadrupole" in kwargs["properties"]:
+            ret["quadrupole"] = [psi_props[mtd + " QUADRUPOLE " + x] for x in ["XX", "XY", "XZ", "YY", "YZ", "ZZ"]]
+        if "mulliken_charges" in kwargs["properties"]:
+            ret["mulliken_charges"] = wfn.get_array("MULLIKEN_CHARGES").np.ravel().tolist()
+        if "lowdin_charges" in kwargs["properties"]:
+            ret["lowdin_charges"] = wfn.get_array("LOWDIN_CHARGES").np.ravel().tolist()
+        if "wiberg_lowdin_indices" in kwargs["properties"]:
+            ret["wiberg_lowdin_indices"] = wfn.get_array("WIBERG_LOWDIN_INDICES").np.ravel().tolist()
+        if "mayer_indices" in kwargs["properties"]:
+            ret["mayer_indices"] = wfn.get_array("MAYER_INDICES").np.ravel().tolist()
+
+        json_data["return_result"] = ret
+    else:
+        raise KeyError("Did not understand Driver key %s." % json_data["driver"])
+
+
+    props = {
+        "calcinfo_nbasis": wfn.nso(),
+        "calcinfo_nmo": wfn.nmo(),
+        "calcinfo_nalpha": wfn.nalpha(),
+        "calcinfo_nbeta": wfn.nbeta(),
+        "calcinfo_natom": mol.geometry().shape[0],
+        "scf_one_electron_energy": psi_props["ONE-ELECTRON ENERGY"],
+        "scf_two_electron_energy": psi_props["TWO-ELECTRON ENERGY"],
+        "nuclear_repulsion_energy": psi_props["NUCLEAR REPULSION ENERGY"],
+        "scf_dipole_moment": [psi_props[x] for x in ["SCF DIPOLE X", "SCF DIPOLE Y", "SCF DIPOLE Z"]],
+        "scf_iterations": int(psi_props["SCF ITERATIONS"]),
+        "scf_total_energy": psi_props["SCF TOTAL ENERGY"],
+    }
+
+    # Pull out optional SCF keywords
+    other_scf = [("DFT VV10 ENERGY", "scf_vv10_energy"), ("DFT XC ENERGY", "scf_xc_energy"),
+                 ("DISPERSION CORRECTION ENERGY", "scf_dispersion_correction_energy")]
+    for pkey, skey in other_scf:
+        if (pkey in psi_props) and (psi_props[pkey] != 0):
+            props[skey] = psi_props[pkey]
+
+    # Write out MP2 keywords
+    if "MP2 CORRELATION ENERGY" in psi_props:
+        props["mp2_same_spin_correlation_energy"] = psi_props["MP2 SAME-SPIN CORRELATION ENERGY"]
+        props["mp2_opposite_spin_correlation_energy"] = psi_props["MP2 OPPOSITE-SPIN CORRELATION ENERGY"]
+        props["mp2_singles_energy"] = 0.0
+        props["mp2_doubles_energy"] = psi_props["MP2 CORRELATION ENERGY"]
+        props["mp2_total_correlation_energy"] = psi_props["MP2 CORRELATION ENERGY"]
+        props["mp2_total_energy"] = psi_props["MP2 TOTAL ENERGY"]
+
+    json_data["properties"] = props
+    json_data["success"] = True
+    print(json.dumps(json_data, indent=2))
+
+    # Reset state
+    optstash.restore()
+    if clean:
+        core.clean()
```

Comment:
do you want either of `psi4.core.clean_options()` or `psi4.core.clean_variables()` here, too?